models/checkout.js

```diff
@@ -0,0 +1,129 @@
+const { query } = require('../db/index');
+
+const Tool = require('./tool.js')
+
+exports.all = async () => {
+  const checkouts = (await query(
+    `SELECT
+      checkouts.id,
+      checkouts."patronName",
+      checkouts."patronContact",
+      checkouts."dueDate",
+      checkouts."toolId",
+      tools."name" as "toolName"
+    FROM "checkouts", tools
+    WHERE checkouts."toolId" = tools.id`,
+  )).rows;
+  return checkouts
+};
+
+exports.create = async properties => {
+  const errors = await validate(properties);
+  if (errors) {
+    return { errors, properties };
+  }
+
+  const createdCheckout = (await query(
+    `INSERT INTO "checkouts"(
+      "patronName",
+      "patronContact",
+      "dueDate",
+      "toolId"
+    ) values ($1, $2, $3, $4) returning *`,
+    [
+      properties.patronName,
+      properties.patronContact,
+      properties.dueDate,
+      properties.toolId,
+    ],
+  )).rows[0];
+  return createdCheckout;
+};
+
+exports.delete = async id => {
+  await query(
+    `DELETE FROM "checkouts"
+      WHERE "id" = $1
+    returning *`,
+    [
+      id,
+    ],
+  );
+  return {};
+};
+
+exports.find = async id => {
+  const checkout = (await query(
+    'SELECT * FROM "checkouts" WHERE "id" = $1 LIMIT 1',
+    [
+      id,
+    ],
+  )).rows[0];
+  return checkout;
+};
+
+exports.update = async newProperties => {
+  const oldProps = await this.find(newProperties.id);
+  const properties = { ...oldProps, ...newProperties };
+
+  const errors = await validate(properties);
+  if (errors) {
+    return { errors, properties };
+  }
+
+  const updatedCheckout = (await query(
+    `UPDATE "checkouts" SET
+    "patronName"=$1,
+    "patronContact"=$2,
+    "dueDate"=$3,
+    "toolId"=$4 WHERE id=$5 RETURNING *`,
+    [
+      properties.patronName,
+      properties.patronContact,
+      properties.dueDate,
+      properties.toolId,
+      properties.id,
+    ],
+  )).rows[0];
+
+  return updatedCheckout;
+};
+
+exports.wherePatronContact = async patronContact => {
```

Comment:
This will be so useful when we want to look up patron history... but might want to standardize data going in later? like if it looks like an email or phone number, recognize and standardize that data before save and then display however we want... let's feel that pain first though I guess